### PR TITLE
DOCS-9118 ServiceNow 6.7

### PR DESCRIPTION
**Pull request by valkyrie69**
Created doc for ServiceNow 6.7

### CODE REVIEW

File: servicenow/6.6/modules/ROOT/pages/servicenow-connector-studio.adoc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 = Use Anypoint Studio to Configure ServiceNow Connector - Mule 4
```
Comment (kevintroller) [inline servicenow/6.6/modules/ROOT/pages/servicenow-connector-studio.adoc:1]:
```suggestion
= Use Anypoint Studio to Configure ServiceNow Connector 6.6 - Mule 4
```

File: servicenow/6.7/modules/ROOT/pages/index.adoc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+* New user
++
+To create your Mule app,
+see xref:servicenow-connector-studio.adoc[Studio Configuration].
+* Advanced user
++
+Read the xref:servicenow-connector-xml-maven.adoc[XML and Maven Support]
+and xref:servicenow-connector-examples.adoc[Examples] topics.
+The Examples topic provides one or more use cases for using the connector.
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/index.adoc]:
```suggestion
* New users, read
** xref:servicenow-connector-studio.adoc[Use Studio to Configure the Connector] to create your Mule app
* Advanced users, read:
** xref:servicenow-connector-xml-maven.adoc[XML and Maven Support]
** xref:servicenow-connector-examples.adoc[Examples], which provides one or more use cases for the connector
```

Comment (kevintroller) [reply servicenow/6.7/modules/ROOT/pages/index.adoc]: Also, I would move the Audiences section afterwards the Common Use Cases for the Connector

File: servicenow/6.7/modules/ROOT/pages/index.adoc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+== Common Use Cases For the Connector
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/index.adoc:23]:
```suggestion
== Common Use Cases for the Connector
```

File: servicenow/6.7/modules/ROOT/pages/index.adoc [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+ServiceNow Connector enables organizations to fully integrate business processes across HR, legal, procurement, operations, marketing, and facilities departments.
+Creating connectivity within and outside the enterprise is quick and simple with connectivity to over 120 Anypoint connectors.
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/index.adoc]:
```suggestion
Creating connectivity within and outside of the enterprise is quick and simple with the integration of over 120 Anypoint Connectors.
```

File: servicenow/6.7/modules/ROOT/pages/index.adoc [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+Use ServiceNow Connector to create instant API connectivity with the ServiceNow API, and quickly and easily interface with ServiceNow from within Anypoint Platform.
+
+
+xref:servicenow-connector-examples.adoc[Example] use cases includes getting a ServiceNow incident record.
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/index.adoc]:
```suggestion
For examples, see xref:servicenow-connector-examples.adoc[example use cases], which include getting a ServiceNow incident record.
```

```suggestion
For examples, see xref:servicenow-connector-examples.adoc[Examples].
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+:page-aliases: connectors::servicenow/servicenow-ex-to-get-incident-table-keys.adoc, connectors::servicenow/servicenow-ex-to-get-record.adoc, connectors::servicenow/servicenow-ex-to-get-records-from-incident-table.adoc, connectors::servicenow/servicenow-XML-reference.adoc, connectors::servicenow/servicenow-connector-examples.adoc
+
+You can try the following examples in Studio or Design Center.
+
+== Retrieve a ServiceNow Incident Record
+
+In these steps, you create a Mule flow in Flow Designer that retrieves a record based on an incident number:
+
+. Create a Mule application as a project in Flow Designer.
+. Add and configure an HTTP Listener as a trigger for your flow, and set a path such as:
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
. Add and configure an HTTP *Listener* as a trigger for your flow, and set the *Path* to `/incident/get`.
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
++
+====
+`/incident/get`
+====
++
+Or, use the default configuration for the HTTP Listener connection.
+
+. Add the `Invoke` operation from the ServiceNow connector to your flow.
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
. Add the ServiceNow *Invoke* operation to your flow.
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+== Retrieve a ServiceNow Incident Record
+
+In these steps, you create a Mule flow in Flow Designer that retrieves a record based on an incident number:
+
+. Create a Mule application as a project in Flow Designer.
+. Add and configure an HTTP Listener as a trigger for your flow, and set a path such as:
++
+====
+`/incident/get`
+====
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion

```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 9]
```diff
@@ -0,0 +1,336 @@
+= ServiceNow Connector 6.7 Examples - Mule 4
+:page-aliases: connectors::servicenow/servicenow-ex-to-get-incident-table-keys.adoc, connectors::servicenow/servicenow-ex-to-get-record.adoc, connectors::servicenow/servicenow-ex-to-get-records-from-incident-table.adoc, connectors::servicenow/servicenow-XML-reference.adoc, connectors::servicenow/servicenow-connector-examples.adoc
+
+You can try the following examples in Studio or Design Center.
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
To fully benefit from using Anypoint Connector for ServiceNow (ServiceNow Connector), see the following examples of configuring the connector both in Anypoint Studio and Design Center.
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 10]
```diff
@@ -0,0 +1,336 @@
...
+. Make these values available as menu items in the *Service* and *Operation* fields.
++
+====
+The names become available once the connector loads the metadata for them. +
+
+IMPORTANT: Do not add these names manually.
+====
++
+. Provide a transformation script for the connector.
+. From the `Invoke` operation in Design Center, click  *Map* for the *Body* field.
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
. From the *Invoke* operation in Design Center, click  *Map* for the *Body* field.
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 11]
```diff
@@ -0,0 +1,336 @@
...
+        inc#number: "INC0012345"
+    }
+}
+----
++
+NOTE: This example changes the `output` data type to `application/xml`.
+
+. Add a Logger to the end of your flow that takes `payload` as the message.
+. Deploy, sync, or run the app.
+. Test the app by opening this from a browser: `+http://url.for.your.app:8081/incident/get+` or from an app such as Postman.
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
. Test the app by opening the URL `+http://url.for.your.app:8081/incident/get+` from a browser  or from an app such as Postman.
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 12]
```diff
@@ -0,0 +1,336 @@
...
+. In the *General* tab in the navigation, set *Service* to `incident` and *Operation* to `getRecords`. +
+. Optionally, you can choose *Show Reference Values* from the values (`ALL`, `TRUE`, `FALSE`) +
++
+IMPORTANT: In Studio 7.5.0 and later you can choose `service` and `operation` keys without specifying the `Show Reference Values` key to resolve the metadata. However, in earlier Studio versions, metadata won’t be loaded until you specify all of the metadata keys, and if you don't specify all of the metadata keys, it results in a tooling exception.
++
+image::servicenow-metadata-without-reference-values.png[Servicenow metadata for Studio 7.5.0 without reference values]
++
+. Set the Message Body to `payload`. +
+
+. Add the Transform Message component to `transform XML to JSON` for better readablity. For example:
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
. Add the *Transform Message* component to `transform XML to JSON` for better readability. For example:
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 13]
```diff
@@ -0,0 +1,336 @@
...
++
+----
+%dw 2.0
+output application/json
+---
+payload
+
+----
++
+. Add a Logger to the end of your flow that takes a `payload` (or `#[payload]`) as the message. +
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
. Add a *Logger* component to the end of your flow that takes a `payload` (or `#[payload]`) as the message. +
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 14]
```diff
@@ -0,0 +1,336 @@
...
++
+image::servicenow-flow-incident-getrecords.png[Servicenow flow - incident get records]
++
+
+. Deploy or Run your app. +
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
. Deploy or run your app. +
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 15]
```diff
@@ -0,0 +1,336 @@
...
+
+For more information, see https://docs.servicenow.com/bundle/orlando-application-development/page/integrate/inbound-soap/concept/return-display-reference-variables.html[ServiceNow - Display value].
+
+== Retrieve ServiceNow Incident Table Keys
+
+This example creates a Mule flow to retrieve keys from the Incident table in a ServiceNow instance.
+
+. Create a Mule application as a project in Anypoint Studio 7. +
+
+. Add and configure an HTTP Listener trigger for your flow using a Path such as this one:
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
. Add and configure an HTTP *Listener* trigger for your flow, and set the *Path* to `/getkeys`.
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 16]
```diff
@@ -0,0 +1,336 @@
...
+
++
+====
+`/getkeys`
+====
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion

```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 17]
```diff
@@ -0,0 +1,336 @@
...
++
+. Add a Transform message component after the HTTP connector.
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
. Add a *Transform Message* component after the HTTP *Listener* operation.
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 18]
```diff
@@ -0,0 +1,336 @@
...
+ns ns0 http://www.service-now.com/incident
+---
+{
+	ns0#getKeys: {
+		ns0#description: 'Test WSDL QA'
+	}
+}
+----
++
+. Add the `Invoke` operation from the ServiceNow connector after the *Transform* component. +
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
. Add the ServiceNow *Invoke* operation after the *Transform Message* component. +
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 19]
```diff
@@ -0,0 +1,336 @@
...
+
+. Set up, test, and save a ServiceNow configuration for the connection to the ServiceNow server. If the connection is unsuccessful, correct any invalid connection parameters and test again. +
+
+. In the *General* tab navigation, set *Service* to `incident` and *Operation* to `getKeys`. +
+
+. Set the Message Body to `payload`. +
+
+. Add the *Transform Message* component to transform XML to JSON for better readablity. For example:
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
. Add the *Transform Message* component to transform XML to JSON for better readability. For example:
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 20]
```diff
@@ -0,0 +1,336 @@
...
+
++
+----
+%dw 2.0
+output application/json
+---
+payload
+----
++
+. Add a Logger to the end of your flow that takes a `payload` (or `#[payload]`) as the message.
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
. Add a *Logger* component to the end of your flow that takes a `payload` (or `#[payload]`) as the message.
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc [hunk 21]
```diff
@@ -0,0 +1,336 @@
...
+
+. Deploy or Run your app.
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-examples.adoc]:
```suggestion
. Deploy or run your app.
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-studio.adoc [hunk 22]
```diff
@@ -0,0 +1,183 @@
...
+
+A source initiates a flow when a specified condition is met.
+You can configure one of these input sources to use with ServiceNow Connector:
+
+* *HTTP > Listener* +
+Initiates a flow each time it receives a request on the configured host and port
+* *Scheduler* +
+Initiates a flow when a time-based condition is met
+
+For example, to configure an HTTP listener, follow these steps:
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-studio.adoc]:
```suggestion
For example, to configure an HTTP *Listener*, follow these steps:
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-studio.adoc [hunk 23]
```diff
@@ -0,0 +1,183 @@
...
+
+. In the *Mule Palette* view, select *HTTP > Listener*.
+. Drag *Listener* to the Studio canvas.
+. On the *Listener* configuration screen, optionally change the value of the *Display Name* field.
+. Specify a value for the *Path* field.
+. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of the HTTP listener in the app.
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-studio.adoc]:
```suggestion
. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of the HTTP *Listener* in the app.
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-studio.adoc [hunk 24]
```diff
@@ -0,0 +1,183 @@
...
+
+[[configure-global-element]]
+== Configure a Global Element for the Connector
+
+When you configure a connector, it’s best to configure a global element that all instances of that connector in the app can use. Configuring a global element requires you to provide the authentication credentials that the connector requires to access the target ServiceNow system. ServiceNow Connector supports Basic and OAuth2 authentication types.
+
+To configure the global element for ServiceNow Connector, follow these steps:
+
+. Select the operation in the Studio canvas.
+. In the configuration screen for the operation, click the plus sign (+) next to the *Connector configuration* field to access the global element configuration fields.
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-studio.adoc]:
```suggestion
. In the configuration screen for the operation, click the plus sign (*+*) next to the *Connector configuration* field to access the global element configuration fields.
```

File: servicenow/6.7/modules/ROOT/pages/servicenow-connector-xml-maven.adoc [hunk 25]
```diff
@@ -0,0 +1,71 @@
...
+</dependency>
+----
+
+Replace `x.x.x` with the version that corresponds to the connector you are using.
+
+To obtain the most up-to-date `pom.xml` file information:
+
+. Go to https://www.mulesoft.com/exchange/[Anypoint Exchange].
+. In Exchange, click *Login* and supply your Anypoint Platform username and password.
+. In Exchange, search for `<connector-name>`.
```
Comment (kevintroller) [inline servicenow/6.7/modules/ROOT/pages/servicenow-connector-xml-maven.adoc]:
```suggestion
. In Exchange, search for `serviceNow`.
```

**Review by kevintroller — DISMISSED** [review]
Minor comments

File: servicenow/6.7/modules/ROOT/pages/index.adoc (hunk 3 above)
Comment (valkyrie69) [reply servicenow/6.7/modules/ROOT/pages/index.adoc:23]: Kim says to use For. I don't remember what she told me but there was a reason.

**Review by kevintroller — APPROVED** [review]

**Pull request merged**